File: backend/src/config/firebase.js
import admin from 'firebase-admin';
import { readFileSync } from 'fs';

let firebaseApp;

export const initializeFirebase = () => {
  try {
    const serviceAccount = JSON.parse(
      readFileSync(process.env.FIREBASE_SERVICE_ACCOUNT_PATH, 'utf8')
    );

    firebaseApp = admin.initializeApp({
      credential: admin.credential.cert(serviceAccount),
    });

    console.log('✅ Firebase Admin initialized');
  } catch (error) {
    console.error('❌ Firebase initialization error:', error.message);
    console.log('⚠️  Continuing without Firebase (download service account key from Firebase Console)');
  }
};

export const getFirebaseAdmin = () => firebaseApp;

File: backend/config/firebase.go
package config

import (
	"context"
	"fmt"
	"log"
	"os"

	firebase "firebase.google.com/go/v4"
	"firebase.google.com/go/v4/auth"
	"google.golang.org/api/option"
)

var FirebaseAuth *auth.Client

// InitializeFirebase initializes Firebase Admin SDK
func InitializeFirebase() {
	serviceAccountPath := os.Getenv("FIREBASE_SERVICE_ACCOUNT_PATH")
	if serviceAccountPath == "" {
		log.Println("⚠️  FIREBASE_SERVICE_ACCOUNT_PATH not set")
		log.Println("⚠️  Continuing without Firebase (download service account key from Firebase Console)")
		return
	}

	opt := option.WithCredentialsFile(serviceAccountPath)
	app, err := firebase.NewApp(context.Background(), nil, opt)
	if err != nil {
		log.Printf("❌ Firebase initialization error: %v\n", err)
		log.Println("⚠️  Continuing without Firebase")
		return
	}

	FirebaseAuth, err = app.Auth(context.Background())
	if err != nil {
		log.Printf("❌ Firebase Auth initialization error: %v\n", err)
		log.Println("⚠️  Continuing without Firebase")
		return
	}

	fmt.Println("✅ Firebase Admin initialized")
}

// GetFirebaseAuth returns the Firebase Auth client
func GetFirebaseAuth() *auth.Client {
	return FirebaseAuth
}
